bowtie/_cli.py

```diff
@@ -880,7 +932,32 @@ def validate(
             for i, instance in enumerate(instances, 1)
         ],
     )
-    return asyncio.run(_run(fail_fast=False, **kwargs, cases=[case]))
+    return asyncio.run(_run_cases(fail_fast=False, **kwargs, cases=[case]))
+
+
+@all_implementations_subcommand()  # type: ignore[reportArgumentType]
+@DIALECT(default=None)
+@LANGUAGE
+async def filter_implementations(
+    implementations: Iterable[Implementation],
+    dialect: Dialect | None,
+    language: _L | None,
+):
+    supporting_implementations: list[str] = []
+    for each in implementations:
+        metadata = each.info.serializable()
+        impl_id = each.info.id
+        impl_name = impl_id.split("/")[-1]
```

Comment:
This too should set off alarm bells that you're parsing something you probably shouldn't. I'm not sure why you're doing the parsing here, this will be completely wrong if someone provides a full image name -- in other words, we treat `foo-bar` as if someone wrote `ghcr.io/bowtie-json-schema/foo-bar`, but someone can provide `foo/bar/baz` and then we start the image from `foo/bar/baz` and make no assumptions about it, whereas you here would print `baz` and therefore not be able to find the image at all.

Reply:
@Julian, as explained by me in the reply to the first comment, the user here isn't really providing the image names.  If you look closely, the `impl_id.split("/")[-1]` I am doing is on the `each.info.id` which is already the full path of `ghcr.io` included image name.